### ISSUE TITLE
remote exectution tests failing with SocketError getaddrinfo: Name or service not known

### ISSUE THREAD

**Issue by pondrejk** [SatelliteQE/robottelo#4578]
Appears both in UI and CLI tests:

```
E               AssertionError: False is not true : host output: Error initializing command #<Proxy::RemoteExecution::Ssh::Dispatcher::Command:0x007fb55408d4f8> SocketError getaddrinfo: Name or service not known Exit status: EXCEPTION
```

**Comment by pondrejk** [comment]
This is given by the current infrastructure for creating hosts, sat can not reach vms from other machine by hostname atm. Looking for a solution, a workaround is to enable the global setting remote_execution_connect_by_ip, executing remote jobs by ip address works all right in this scenario.

**Comment by pondrejk** [comment]
The problem should be fixed by creating the vm host with the right bridge interface. Filed #4624 to some testing for remote_execution_connect_by_ip

**Comment by sghai** [comment]
At first place, this is the issue w/ hostname resolution. Just a give a brief background, we are trying to create a VM through `robottelo.vm` that provides the hostname like `140167802457360.DOMAIN` and later we run the REX job on that VM and it failed w/

```
2017-04-27 08:46:51 - robottelo.ssh - INFO - >>> LANG=en_US.UTF-8  hammer -v -u admin -p changeme  job-invocation output --host="140167802457360.lab.eng.rdu2.redhat.com" --id="3"
2017-04-27 08:46:53 - robottelo.ssh - INFO - <<< stdout
Error initializing command #<Proxy::RemoteExecution::Ssh::Dispatcher::Command:0x007f23a0191a20>
SocketError getaddrinfo: Name or service not known
Exit status: EXCEPTION
```

**Comment by sghai** [comment]
This comment clearly explains everything and w/ a workaround of DDNS..
https://github.com/SatelliteQE/robottelo/issues/4101#issuecomment-272400033

I can't think of any other solution. Please respond if anyone has any other suggestion/solution.

**Comment by sghai** [comment]
Complete implementation of DDNS is defined here though not sure how flexible it would be w/ no of hosts..

https://github.com/SatelliteQE/robottelo/blob/6.2.z/robottelo/vm_capsule.py

**Comment by san7ket** [comment]
vm.py will give us the ip and hostname of created host , we can always add those entry in satellite's /etc/hosts file , a ugly solution , but it should be inexpensive. 
We can come up with some module to handle this, along with the "clean-up".

**Comment by rplevka** [comment]
@sghai @san7ket,
our `libvirt1` libvirt instance used for CR has all the required vlans configured. This means, i was successful in creating a vm using `vm.py` on it, but... this requires a satellite you're working with to be in one of these vlans too, which does not apply to the bkr machines.
So this will definitely work with our automation since the testing boxes already are a part of this infra (woth all the required vlans), but it will be tricky to use it to test things "off the record".
I think we have a box called 'sat6-feature' designated to tests like these. So i think we should stick to this design.

**Comment by san7ket** [comment]
@rplevka So, the vm's provisioned `--host="140167802457360.lab.eng.rdu2.redhat.com" `is provisioned on `libvirt1` , with the help of `vm.py`. While doing so, is the VM that is being created using the same Vlan that the satellite is in.

**Comment by rplevka** [comment]
@san7ket, no, you got to specify it using the `bridge` parameter, @pondrejk has recently implemented to `vm.py`

**Comment by rplevka** [comment]
blocked by: https://github.com/SatelliteQE/automation-tools/issues/616

**Comment by rplevka** [comment]
this requires another action inside the libvirt1 resource - the avahi daemon needs to be setup in order to `vm.py` being able to ping the vm's hostname by `<vm>.local`.
@lpramuk is working on this

**Comment by sghai** [comment]
@lpramuk : are we closer to fix this issue as per comments https://github.com/SatelliteQE/robottelo/issues/4578#issuecomment-317985109 ?

**Comment by san7ket** [comment]
@lpramuk still failing , is it due to subnet/IP issue ?

**Comment by rplevka** [comment]
The infrastructure issues have been fixed (REX now works at least over IP).

there's another issue with REX over hostname, howver this should be covered in different issue.

**Comment by san7ket** [comment]
This is still reproducible on upgrade boxes,  
```
for i in {1..60}; do ping -c1 140188538972368.local && exit 0; sleep 1; done; exit 1
2017-09-12 18:35:41 - robottelo.ssh - INFO - <<< stderr
ping: 140188538972368.local: Name or service not known
```
unable to ping `vm.<local>`

Was there something added to boxes to make this work ? that must be missing on upgrade images

**Comment by rplevka** [comment]
@san7ket I think upgrade boxes uses a different infra - they are VM so i guess there is no vlan plumbing set up - @lpramuk i know, you have insight into this, can you confirm?

**Comment by rplevka** [comment]
@san7ket also, i've noticed that UI ones still use the 

settings.clients.provisioning_server
to provision the VM's on, which don't exist in the same subnet, so they obviously can't work

**Comment by san7ket** [comment]
@rplevka these are CLI tests that failed in last snap, and upgrade boxes have their own subnet , upgrade_el6 and upgrade_el7.
@lpramuk anything apart from above was done ??  if anything was added to automation-tools or CI , will be missed in upgrade boxes/infra.

**Comment by san7ket** [comment]
@rplevka @lpramuk 
CLI test's for REX are still failing for latest snap on upgrade, Reopening the issue for discussion
Output from one of the test.
tests.foreman.cli.test_remoteexecution.RemoteExecutionTestCase.test_positive_install_multiple_packages_with_a_job
```
Creating snapshot guest 140415107801488.satellite.~~domain~~.com
Creating snapshot image
Formatting '/opt/robottelo/images/140415107801488.satellite.~~domain~~.com.img', fmt=qcow2 size=214748364800 backing_file='/var/lib/libvirt/images/rhel73-base.img' encryption=off cluster_size=65536 lazy_refcounts=off 
OS: linux rhel 7 3
Disable fsck startup check for all volumes
Configuring message of the day
Configuring /etc/hosts file
Setting up for Fedora / RHEL
Setting MAC address
Setting hostname
Provisioning guest 140415107801488.satellite.~~domain~~.com
Hostname: 140415107801488.satellite.~~domain~~.com
CPUs:     1
Memory:   512 MB
Swap:     512 MB
MAC:      52:54:00:52:fb:cf

Starting install...

Creating domain...                                          |    0 B  00:00     
Domain creation completed.

2017-09-24 02:29:55 - robottelo.ssh - INFO - <<< stderr
WARNING  No operating system detected, VM performance may suffer. Specify an OS with --os-variant for optimal results.

2017-09-24 02:29:55 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7fb4f09f2850
2017-09-24 02:29:55 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7fb4f1e06c10
2017-09-24 02:29:55 - robottelo.ssh - INFO - Connected to [libvirt1.com]
2017-09-24 02:29:55 - robottelo.ssh - INFO - >>> for i in {1..60}; do ping -c1 140415107801488.local && exit 0; sleep 1; done; exit 1
2017-09-24 02:35:57 - robottelo.ssh - INFO - <<< stderr
ping: 140415107801488.local: Name or service not known
ping: 140415107801488.local: Name or service not known
ping: 140415107801488.local: Name or service not known
ping: 140415107801488.local: Name or service not known
ping: 140415107801488.local: Name or service not known
ping: 140415107801488.local: Name or service not known
ping: 140415107801488.local: Name or service not known
```

**Comment by san7ket** [comment]
This seems to be related to https://github.com/SatelliteQE/satellite6-upgrade/issues/130
